hazelcast/src/main/java/com/hazelcast/query/Predicates.java

```diff
@@ -29,77 +29,342 @@
+ * never pass such items.
+ * <p>
+ * <b>Type Conversion</b>
+ * <p>
+ * If the type of the stored value doesn't match the type of the value provided to the predicate, type conversion is
+ * performed before predicate evaluation. The provided value is converted to match the type of the stored attribute value.
+ * If there is no conversion exists to match the type, {@link IllegalArgumentException} is thrown.
+ * <p>
+ * Depending on the attribute type following conversions may apply:
+ * <ul>
+ * <li>{@link Number Numeric types}
+ * <ul>
+ * <li>Strings are parsed in attempt to extract the represented numeric value from them in the same way as
+ * {@link Integer#parseInt(String)} and analogous methods of other types do. Strings containing no meaningful
+ * representation will produce {@link NumberFormatException}.
+ * <li>Widening conversion may be performed in the same way as described in JLS 5.1.2 Widening Primitive Conversions.
+ * <li>Narrowing conversion may be performed in the same way as described in JLS 5.1.3 Narrowing Primitive Conversions.
+ * </ul>
+ * <li>{@link Boolean} type
+ * <ul>
+ * <li>A string that case-insensitively equals to {@code "true"} is converted to {@code true}, all other strings are
+ * converted to {@code false}.
+ * <li>Any non-zero numeric value is converted to {@code true}, the zero is converted to {@code false}.
+ * </ul>
+ * <li>String type
+ * <ul>
+ * <li>{@link Object#toString()} is invoked to produce the string representation for non-string values.
+ * </ul>
+ * <li>{@link Character} type
+ * <ul>
+ * <li>The first character of a string is used for the conversion. Empty strings are not allowed and will produce
+ * {@link IllegalArgumentException}.
+ * <li>Any numeric value is converted to an integer value representing a single UTF-16 code unit to create a character
+ * from. This process may involve widening and narrowing conversions as described in JLS 5.1.2 Widening Primitive
+ * Conversions and 5.1.3 Narrowing Primitive Conversions.
+ * </ul>
+ * <li>Enum types
+ * <ul>
+ * <li>Any non-string value is converted to a string using {@link Object#toString()}, which is interpreted as a
+ * case-sensitive enum member name.
+ * </ul>
+ * <li>{@link java.math.BigInteger BigInteger} type
+ * <ul>
+ * <li>Numeric values are converted to {@link java.math.BigInteger BigInteger} by applying widening or narrowing
+ * conversion as described in JLS 5.1.2 Widening Primitive Conversions and 5.1.3 Narrowing Primitive Conversions.
+ * <li>Boolean {@code true} and {@code false} are converted to {@link java.math.BigInteger#ONE BigInteger.ONE}
+ * and {@link java.math.BigInteger#ZERO BigInteger.ZERO} respectively.
+ * <li>A value of any other type is converted to string using {@link Object#toString()}, which is interpreted as
+ * a base 10 representation. If the representation is invalid, {@link NumberFormatException} will be thrown.
```

Comment:
Isn't this using `BigInteger(String)` constructor? Or is it using custom parsing. If so, it would be better to refer to that.

Reply:
fixed